File: QViz.Java/src/main/java/com/periscope/qviz/json/TestCaseGUI.java
package com.periscope.qviz.json;

import java.util.ArrayList;
import java.util.List;

/**
 * JSON Model for GUI Test Case Details
 */
public class TestCaseGUI extends TestCase {

	/**
	 * List of Test Case Step Objects associated with the Test Case
	 */
	public List<TestCaseStep> testCaseSteps;

	/**
	 * Lise of Test Action Objects associated with the Test Case
	 */
	public List<Action> testActions;

	/**
	 * List of Test API Objects associated with the Test Case
	 */
	public List<TestAPI> testAPIs;

	/**
	 * Default Constructor
	 */
	public TestCaseGUI() {
		super();
		this.testCaseSteps = new ArrayList<>();
		this.testActions = new ArrayList<>();
		this.testAPIs = new ArrayList<>();
	}

}

File: QViz.NET/Periscope.QViz/JSON/TestCaseGUI.cs
﻿using System.Collections.Generic;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for GUI Test Case Details
	/// </summary>
	public class TestCaseGUI : TestCase
	{
		/// <summary>
		/// List of Test Case Step Objects associated with the Test Case
		/// </summary>
		public IList<TestCaseStep> testCaseSteps;
		
		/// <summary>
		/// GET List of Test Action Objects associated with the Test Case
		/// </summary>
		public IList<TestAction> testActions { get; }

		/// <summary>
		/// POST List of Test Action Objects associated with the Test Case
		/// </summary>
		public IList<Action> testCaseActions;
		/// <summary>
		/// List of Test API Objects associated with the Test Case
		/// </summary>
		public IList<TestAPI> testAPIs;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public TestCaseGUI() : base()
		{
			this.testCaseSteps = new List<TestCaseStep>();
			this.testActions = new List<TestAction>();
			this.testAPIs = new List<TestAPI>();
			this.testCaseActions = new List<Action>();
		}
	}
}